ansible_collections/arista/avd/roles/eos_validate_state/tasks/main.yml

```diff
@@ -0,0 +1,281 @@
+---
+
+    - name: include device intended structure configuration variables
+      include_vars: '{{ inventory_dir }}/intended/structured_configs//{{ inventory_hostname }}.yml'
+      delegate_to: localhost
+      tags:
+        - always
+
+    - name: run show version on EOS devices
+      eos_command:
+        commands: "show version | json"
+      register: eos_version
+      tags:
+        - platform_information
+
+    - name: Display device platform and release information
+      debug:
+        msg: "The device {{ inventory_hostname }} is a {{ eos_version.stdout_lines.0.modelName }} model running EOS version {{ eos_version.stdout_lines.0.version }}"
+      tags:
+        - platform_information
+
+    - name: collect power supplies status
+      eos_command:
+        commands: "show system environment power | json"
+      register: power
+      tags:
+        - hardware
+
+    - name: validate power supplies status
+      assert:
+        that:
+          - item.value.state == 'ok'
+        quiet: true
+      loop: "{{ power.stdout_lines.0.powerSupplies | dict2items }}"
+      ignore_errors: "{{ validation_mode_loose }}"
+      tags:
+        - hardware
+
+    - name: collect fan status
+      eos_command:
+        commands: "show system environment cooling | json"
+      register: fan
+      tags:
+        - hardware
+
+    - name: validate fan status (power supplies)
+      assert:
+        that:
+          - item.status == 'ok'
+        quiet: true
+      loop: "{{ fan.stdout_lines.0.powerSupplySlots }}"
+      ignore_errors: "{{ validation_mode_loose }}"
+      tags:
+        - hardware
+
+    - name: validate fan status (fan tray)
+      assert:
+        that:
+          - item.status == 'ok'
+        quiet: true
+      loop: "{{ fan.stdout_lines.0.fanTraySlots }}"
+      ignore_errors: "{{ validation_mode_loose }}"
+      tags:
+        - hardware
+
+    - name: validate temperature
+      eos_command:
+        commands: "show system environment temperature | json"
+        wait_for: result[0].systemStatus eq 'temperatureOk'
+      ignore_errors: "{{ validation_mode_loose }}"
+      tags:
+        - hardware
+
+    - name: collect transceivers inventory
+      eos_command:
+        commands: "show inventory | json"
+      register: inventory
+      tags:
+        - hardware
+
+    - name: validate transceivers manufacturers
+      assert:
+        that:
+          - "item.value.mfgName in ['Not Present', 'Arastra, Inc.', 'Arista Networks']"
+        quiet: true
+      loop: "{{ inventory.stdout_lines.0.xcvrSlots | dict2items }}"
+      ignore_errors: "{{ validation_mode_loose }}"
+      tags:
+        - hardware
+
+    - name: validate reload cause
+      eos_command:
+        commands: "show reload cause | json"
+        wait_for: result[0].resetCauses[0].description eq 'Reload requested by the user.'
+        retries: 2
+        interval: 2
+      ignore_errors: "{{ validation_mode_loose }}"
+      tags:
+        - optional
```

Comment:
Maybe using `optional` and [`never`](https://docs.ansible.com/ansible/latest/user_guide/playbooks_tags.html#special-tags) would be more ansible standard as it is skipped by default and make it really optional with explicit definition during playbook execution

Reply:
yes, will add the - never tag!